apps/gallery/test/marionette/lib/gallery.js

```diff
@@ -310,6 +386,45 @@ Gallery.prototype = {
     }.bind(this));
   },
 
+  switchToGalleryFrame: function() {
+    var galleryFrame = this.client.findElement('iframe[src*="' +
+                                          Gallery.ORIGIN + '"]');
+    this.waitFor(galleryFrame);
```

Comment:
I am equally confused, I have removed switchToGalleryFrame and now using single function switchToApp inside system.js for switching between apps and it works great. I am trying to remember why I needed to switch using selector but failed to replicate it.